### ISSUE TITLE
[GH Actions] Mettre le token pour Expo

### ISSUE THREAD

**Issue by godetremy** [PapillonApp/papillon-v7#225]
Il faut mettre un token expo lié au projet Papillon pour pouvoir voir les PR.

**Comment by Gabriel29306** [comment]
À part erreur de ma part, il y avait déjà eu une issue sur cet aspect mais que le forfait gratuit de eas est trop limité (30 builds par mois)

**Comment by Louis454545** [comment]
Yes et aussi on a pas voulu mon idée pour faire un gh action qui dure indéfiniment

**Comment by godetremy** [comment]
> À part erreur de ma part, il y avait déjà eu une issue sur cet aspect mais que le forfait gratuit de eas est trop limité (30 builds par mois)

Non c'est pas une erreur, mais juste la docs d'expo qui est pas très bien fournie sur ce contexte ! Donc je vais résumer en gros ce que j'ai lu : 

Le principe de faires des previews sur expo n'est pas reconnu comme un build. D'ailleurs ce n'est pas expo en lui même mais EAS (un de leurs services). EAS propose de multiples fonctions, dont le build en ipa / apk etc. C'est le service EAS build. Le service qui nous intéresse c'est le service EAS updates qui a des contraintes différentes qui sont les suivantes : 

- Les builds ne doivent pas durer plus de 45min (mais je ne crois meme pas que les GitHub Actions puissent durer si longtemps)
- Les bundles sont limités à 30mo
- Les liens expirent au bout de 30j

Si une de ces conditions ne fonctionne pas au moment du build, le terminal informe l'utilisateur que le build a échouer et s'arrête automatiquement si aucun abonnement n'est lancer.

On va faire quelque test en interne, mais selon la doc, tout est ok ! 👌

En revanche toi, Louis, tu as du te tromper de doc (elle est foireuse je te l'accorde). Ton GH action publier la mises à jour sur les stores à chaque PR, ce qui n'est pas vraiment ce qu'on voulait !

**Comment by Gabriel29306** [comment]
Concernant les actions GitHub, voici les limites (tirées de la [doc](https://docs.github.com/en/enterprise-cloud@latest/actions/administering-github-actions/usage-limits-billing-and-administration)):

- **Durée d'exécution des tâches** : chaque tâche d'un workflow peut s'exécuter pendant une durée maximale de 6 heures. Si une tâche atteint cette limite, elle est interrompue et ne parvient pas à se terminer.

- **Durée d'exécution du workflow** - Chaque exécution du workflow est limitée à 35 jours. Si une exécution de workflow atteint cette limite, elle est annulée. Cette période comprend la durée d'exécution et le temps passé à attendre et à approuver.

**Comment by Louis454545** [comment]
> Concernant les actions GitHub, voici les limites (tirées de la [doc](https://docs.github.com/en/enterprise-cloud@latest/actions/administering-github-actions/usage-limits-billing-and-administration)):
> 
> - **Durée d'exécution des tâches** : chaque tâche d'un workflow peut s'exécuter pendant une durée maximale de 6 heures. Si une tâche atteint cette limite, elle est interrompue et ne parvient pas à se terminer.
> 
> - **Durée d'exécution du workflow** - Chaque exécution du workflow est limitée à 35 jours. Si une exécution de workflow atteint cette limite, elle est annulée. Cette période comprend la durée d'exécution et le temps passé à attendre et à approuver.

Un workflow sur une pr dure 4h max

**Comment by Louis454545** [comment]
> > À part erreur de ma part, il y avait déjà eu une issue sur cet aspect mais que le forfait gratuit de eas est trop limité (30 builds par mois)
> 
> Non c'est pas une erreur, mais juste la docs d'expo qui est pas très bien fournie sur ce contexte ! Donc je vais résumer en gros ce que j'ai lu : 
> 
> Le principe de faires des previews sur expo n'est pas reconnu comme un build. D'ailleurs ce n'est pas expo en lui même mais EAS (un de leurs services). EAS propose de multiples fonctions, dont le build en ipa / apk etc. C'est le service EAS build. Le service qui nous intéresse c'est le service EAS updates qui a des contraintes différentes qui sont les suivantes : 
> 
> - Les builds ne doivent pas durer plus de 45min (mais je ne crois meme pas que les GitHub Actions puissent durer si longtemps)
> - Les bundles sont limités à 30mo
> - Les liens expirent au bout de 30j
> 
> Si une de ces conditions ne fonctionne pas au moment du build, le terminal informe l'utilisateur que le build a échouer et s'arrête automatiquement si aucun abonnement n'est lancer.
> 
> On va faire quelque test en interne, mais selon la doc, tout est ok ! 👌
> 
> En revanche toi, Louis, tu as du te tromper de doc (elle est foireuse je te l'accorde). Ton GH action publier la mises à jour sur les stores à chaque PR, ce qui n'est pas vraiment ce qu'on voulait !

Non, je n'ai pas fait de pr pour mon truc, pour avoir un workflow qui dure indéfiniment, il y avait 2 workflow qui se lançaient mutuellement

**Comment by Louis454545** [comment]
> > À part erreur de ma part, il y avait déjà eu une issue sur cet aspect mais que le forfait gratuit de eas est trop limité (30 builds par mois)
> 
> Non c'est pas une erreur, mais juste la docs d'expo qui est pas très bien fournie sur ce contexte ! Donc je vais résumer en gros ce que j'ai lu : 
> 
> Le principe de faires des previews sur expo n'est pas reconnu comme un build. D'ailleurs ce n'est pas expo en lui même mais EAS (un de leurs services). EAS propose de multiples fonctions, dont le build en ipa / apk etc. C'est le service EAS build. Le service qui nous intéresse c'est le service EAS updates qui a des contraintes différentes qui sont les suivantes : 
> 
> - Les builds ne doivent pas durer plus de 45min (mais je ne crois meme pas que les GitHub Actions puissent durer si longtemps)
> - Les bundles sont limités à 30mo
> - Les liens expirent au bout de 30j
> 
> Si une de ces conditions ne fonctionne pas au moment du build, le terminal informe l'utilisateur que le build a échouer et s'arrête automatiquement si aucun abonnement n'est lancer.
> 
> On va faire quelque test en interne, mais selon la doc, tout est ok ! 👌
> 
> En revanche toi, Louis, tu as du te tromper de doc (elle est foireuse je te l'accorde). Ton GH action publier la mises à jour sur les stores à chaque PR, ce qui n'est pas vraiment ce qu'on voulait !

Et si tu parles de cette pr : https://github.com/PapillonApp/Papillon/pull/183 savais juste pas lire, il ne publiait jamais sur les stores quand on le lançaient via pr
